firmware/application/apps/ui_playlist.cpp

```diff
@@ -146,7 +146,7 @@ void PlaylistView::toggle() {
         track_number = 0;
         playlist_db.clear();
         playlist_masterdb.clear();
-    } else {
+    } else if (!thread_null || (!playlist_db.empty() || !playlist_masterdb.empty())) {
```

Comment:
This condition doesn't seem right to me. OR doesn't seem like the right thing. What are you trying to test here?